languages.toml

```diff
@@ -319,6 +319,19 @@ indent = { tab-width = 2, unit = "  " }
 name = "tsx"
 source = { git = "https://github.com/tree-sitter/tree-sitter-typescript", rev = "3e897ea5925f037cfae2e551f8e6b12eec2a201a", subpath = "tsx" }
 
+[[language]]
+name = "jsx"
+scope = "source.jsx"
+injection-regex = "^(jsx)$"
+file-types = ["jsx"]
+roots = []
+language-server = { command = "typescript-language-server", args = ["--stdio"], language-id = "javascriptreact" }
+indent = { tab-width = 2, unit = "  " }
+
+[[grammar]]
+name = "jsx"
+source = { git = "https://github.com/tree-sitter/tree-sitter-typescript", rev = "3e897ea5925f037cfae2e551f8e6b12eec2a201a", subpath = "jsx" }
```

Comment:
There is no `jsx` subpath and that rev doesn't exist

Reply:
Appologies, I am not very familiar with those libs

> You should be able to just change the "javascript" definition to say `file-types = ["js", "jsx", "mjs"]`

Would that be enough to fix the issue? No need to add a dedicated section? I though I may use typescriptreact because it's compatible with javascript

Reply:
Yeah it's a bit weird, but tree-sitter-javascript supports both `js` and `jsx`, but tree-sitter-typescript requires you to build two separate grammars.